include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -553,14 +559,19 @@ __pattern_adjacent_find(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _R
     auto __rng1 = oneapi::dpl::__ranges::take_view_simple(__rng, __rng.size() - 1);
     auto __rng2 = oneapi::dpl::__ranges::drop_view_simple(__rng, 1);
 
+    using __size_calc = oneapi::dpl::__ranges::__first_size_calc;
+
+    assert(__rng1.size() == __rng2.size());
+
     // TODO: in case of conflicting names
     // __par_backend_hetero::make_wrapped_policy<__par_backend_hetero::__or_policy_wrapper>()
```

Comment:
I think we should be able to remove this TODO if you agree with my above assessment.  In any case, the comment needs to at least change because `__or_policy_wrapper` no longer exists.  This exists in a number of places.

Reply:
removed